### PR TITLE
Feature/beth

### PR DESCRIPTION
**Pull request by bmorris-revacomm**

### CODE REVIEW

File: Logging/HelperFiles/PBLButton.swift [hunk 1]
```diff
@@ -8,39 +8,36 @@
...
         self.setTitleColor(.slate, for: .normal)
-        self.updateFont(to: FontNames.dmSansBold)
-        self.titleLabel?.adjustsFontSizeToFitWidth = true
-        //self.titleLabel?.adjustsFontForContentSizeCategory = true
-    }
-    
-    func updateFont(to font: String) {
-        guard let size = self.titleLabel?.font.pointSize else { return }
-        self.titleLabel?.font = UIFont(name: font, size: size)
+        self.titleLabel?.font = UIFont(name: FontNames.dmSansRegular, size: UIScreen.main.bounds.height/56.9)
```
Comment (Pete-Hoch) [inline Logging/HelperFiles/PBLButton.swift:19]: I realize that this is targeted at the iPad, but perhaps we should have a min and max value for the font size? Maybe an extension to UIFont?

Also, what does 56.9 mean? I refer to these as 'magic numbers'. It's just a constant with no context. Do we really want a different size font in landscape vs portrait?

Minimum:

let SCALE_FONT_TO_SCREEN_SIZE = 56.9

Then use the constant. And a comment with a deeper explanation is even better.

File: Logging/HelperFiles/PBLButton.swift [hunk 2]
```diff
@@ -8,39 +8,36 @@
...
-        self.heightAnchor.constraint(equalToConstant: 64.0).isActive = true
-        self.frame.size.height = 64
-        self.contentEdgeInsets = UIEdgeInsets(top: 0, left: 20, bottom: 0, right: 20)
+//        print("Screen height: \(UIScreen.main.bounds.height)")
+//        print("Screen width: \(UIScreen.main.bounds.width)")
+//        print("Button height: \(UIScreen.main.bounds.height/21.3)")
+//        print("Button font size: \(UIScreen.main.bounds.height/68.3)")
+        
+        ///Divided by 21.3 because that is aproximately 64px in height for the 12.9in ipad
+        self.heightAnchor.constraint(equalToConstant: UIScreen.main.bounds.height/21.3).isActive = true
```
Comment (Pete-Hoch) [inline Logging/HelperFiles/PBLButton.swift:33]: Yes, comment like this is good. But why is this better than just using a 64p font?

Plus there are a lot more magic numbers that follow.

File: Logging/HelperFiles/PBLButton.swift [hunk 3]
```diff
@@ -72,11 +67,18 @@ class PBLOverviewButton: PBLButtonClear {
...
         self.layer.shadowOffset = CGSize(width: 0.0, height: 2.0)
         self.layer.shadowRadius = 2
     }
+}
+
+class PBLPrintButton: PBLButtonClear {
     
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        self.titleLabel?.font = UIFont(name: FontNames.dmSansBold, size: UIScreen.main.bounds.height/97.6)
```
Comment (Pete-Hoch) [inline Logging/HelperFiles/PBLButton.swift:82]: Some understanding go magic number needed.

**Review by Pete-Hoch — CHANGES_REQUESTED** [review]
Mostly comments, but we should clean up some of this.

File: Logging/Controllers/View Controllers/Form 781/FlightListViewController.swift [hunk 4]
```diff
@@ -297,30 +359,22 @@ class FlightListViewController: UIViewController {
     }
     
     @IBAction func newFlightButtonTapped(_ sender: UIButton) {
-        guard let form = Form781Controller.shared.forms.last else { return }
+        guard let form = Form781Controller.shared.forms.last else { return Alerts.showNoFormAlert(on: self) }
```
[the remark this answers is not in the document]
Comment (petermisik) [reply Logging/Controllers/View Controllers/Form 781/FlightListViewController.swift:362]: I'll make sure to add this to my toolbox.

File: Logging/Controllers/View Controllers/Form 781/FlightListViewController.swift [hunk 5]
```diff
@@ -466,47 +520,186 @@ class FlightListViewController: UIViewController {
...
     
-    func editButtonTapped(cell: FlightTableViewCell) {
+    func presentAircrewInputErrorAlert() {
+        guard let form = Form781Controller.shared.forms.last,
+              let lastName = lastName.text,
+              let firstName = firstName.text,
+              let ssn = ssn.text,
+              let flightAuthDutyCode = flightAuthDutyCode.text,
+              let flyingOrigin = flyingOrigin.text
+        else { return }
```
[the remark this answers is not in the document]
Comment (petermisik) [reply Logging/Controllers/View Controllers/Form 781/FlightListViewController.swift:564]: Should we add this as a bug in Jira?

File: Logging/Controllers/View Controllers/Form 781/FlightListViewController.swift [hunk 6]
```diff
@@ -466,47 +520,186 @@ class FlightListViewController: UIViewController {
...
+        }
     }
     
-    func deleteButtonTapped(cell: FlightTableViewCell) {
+    func highlightAircrew() {
+        lastName.text == "" ? Helper.highlightRed(textField: lastName) : Helper.unhighlight(textField: lastName)
+        firstName.text == "" ? Helper.highlightRed(textField: firstName) : Helper.unhighlight(textField: firstName)
+        ssn.text == "" ? Helper.highlightRed(textField: ssn) : Helper.unhighlight(textField: ssn)
+        flightAuthDutyCode.text == "" ? Helper.highlightRed(textField: flightAuthDutyCode) : Helper.unhighlight(textField: flightAuthDutyCode)
+        flyingOrigin.text == "" ? Helper.highlightRed(textField: flyingOrigin) : Helper.unhighlight(textField: flyingOrigin)
```
[the remark this answers is not in the document]
Comment (petermisik) [reply Logging/Controllers/View Controllers/Form 781/FlightListViewController.swift:581]: I'll work on getting this added to Helperfunction.swift

**Review by petermisik — COMMENTED** [review]
Take a look at Pete's comments and let me know where I can help.

**Pull request merged**